lib/seam.rb

```diff
@@ -5,70 +5,18 @@
-require_relative "seam/resources/acs_entrance"
-require_relative "seam/resources/acs_system"
-require_relative "seam/resources/acs_user"
-require_relative "seam/resources/action_attempt"
-require_relative "seam/resources/client_session"
-require_relative "seam/resources/climate_setting_schedule"
-require_relative "seam/resources/connect_webview"
-require_relative "seam/resources/connected_account"
-require_relative "seam/resources/device"
-require_relative "seam/resources/device_provider"
-require_relative "seam/resources/enrollment_automation"
-require_relative "seam/resources/event"
-require_relative "seam/resources/network"
-require_relative "seam/resources/noise_threshold"
-require_relative "seam/resources/phone"
-require_relative "seam/resources/service_health"
-require_relative "seam/resources/unmanaged_access_code"
-require_relative "seam/resources/unmanaged_device"
-require_relative "seam/resources/user_identity"
-require_relative "seam/resources/webhook"
-require_relative "seam/resources/workspace"
-require_relative "seam/clients/access_codes"
-require_relative "seam/clients/access_codes_simulate"
-require_relative "seam/clients/access_codes_unmanaged"
-require_relative "seam/clients/acs_access_groups"
-require_relative "seam/clients/acs"
-require_relative "seam/clients/acs_credential_pools"
-require_relative "seam/clients/acs_credential_provisioning_automations"
-require_relative "seam/clients/acs_credentials"
-require_relative "seam/clients/acs_entrances"
-require_relative "seam/clients/acs_systems"
-require_relative "seam/clients/acs_users"
-require_relative "seam/clients/action_attempts"
-require_relative "seam/clients/client_sessions"
-require_relative "seam/clients/connect_webviews"
-require_relative "seam/clients/connected_accounts"
-require_relative "seam/clients/devices"
-require_relative "seam/clients/devices_simulate"
-require_relative "seam/clients/devices_unmanaged"
-require_relative "seam/clients/events"
-require_relative "seam/clients/locks"
-require_relative "seam/clients/networks"
-require_relative "seam/clients/noise_sensors_noise_thresholds"
-require_relative "seam/clients/noise_sensors"
-require_relative "seam/clients/noise_sensors_simulate"
-require_relative "seam/clients/phones"
-require_relative "seam/clients/phones_simulate"
-require_relative "seam/clients/thermostats_climate_setting_schedules"
-require_relative "seam/clients/thermostats"
-require_relative "seam/clients/user_identities"
-require_relative "seam/clients/user_identities_enrollment_automations"
-require_relative "seam/clients/webhooks"
-require_relative "seam/clients/workspaces"
+require_relative "seam/base_client"
+require_relative "seam/base_resource"
+
+require_relative "seam/routes/resources/resource_errors_support"
+require_relative "seam/routes/resources/resource_warnings_support"
+
+Dir["#{__dir__}/seam/routes/resources/*.rb"].each do |file|
+  next if file.end_with?("resource_errors_support.rb", "resource_warnings_support.rb")
```

Comment:
Can we move these up one level? What uses the `_support` files?

Reply:
`_support` classes are included in every resource class that has `errors` or `warnings` properties. By including these support classes, resource classes gain access to `errors` and `warnings` getters that return an array, where each error or warning object is converted into a `ResourceWarning` or `ResourceError` class instance.

> Can we move these up one level?
Do you mean moving `lib/seam.rb` to the root level `seam.rb`? 

I was also thinking to move all theses resource and client requires to their respective `seam/routes/clients/index.rb` and `seam/routes/resources/index.rb` files and then just require index files here.

Reply:
`index.rb` is not a common ruby convention I'm aware of (maybe it goes by a different name, like `__init__.py` in python?)

Anyway, this dynamic is require is a good approach if we didn't know what we might find there, but since we generate everything I agree we should generate the imports as you suggested.

Reply:
I found cases of `main.rb` or `loader.rb` but not sure if that can be used with similar intent

Reply:
Ok We can just use `index.rb` for now :)